src/Dtos.cs

```diff
@@ -102,4 +102,5 @@ public record ContentToCopy(string SourcePath, string TargetPath);
 /// </summary>
 /// <param name="Url"></param>
 /// <param name="OutputFile"></param>
-public record PageToGenerate(string Url, string OutputFile);
+/// <param name="OriginalFile">the full path of the original file</param>
```

Comment:
>`lastmod` node isn't about file time generation, but about release date or modified date.

yes it's for the date the page was modified at, and this is taken from `PageToGenerate.OriginalFile` whic is the markdown file not the generated html file.
i chose to do it this way to make it easier for the user.

> maybe we should consider to include ModifiedDateTime metadata to distinguish between Published and Modified

that will mean we have to add another property in `IFrontMatter` or manually parse the frontmatter to find the `LastModified` property.

> I have few ideas

(1) sounds very easy it's tempting honestly.
(2) we can distinguish between pages and blog posts if we keep the new property i added which can only have a value if it's a blog post else it would be null.
(3) is very complicated just to allow for a `modified date`, maybe we just change `OriginalFile` from a string to `FileInfo` (or a custom class/record) and avoid having to add new properties to the record if we need something else in the future.

in my view `PagesToGenerate` should only have data about the file that's being generated

Reply:
It has to be the `PublishedDate` what gets into the `lastmod`. 

So the (1) is the way. Wrap it to a class. `SitemapInfo` is exactly what it is, but disqualifies any future addition that might be needed. What about `AdditionaFileInfo`? (`FileInfo` is already a class)... probably sufficiently generic...

I feel in my bones we'll revisit this one... (Hello future me!)

Reply:
should we really use the publish date instead of modify date? because the protocol explicitly states it should be "*the date the linked page was last modified, not when the sitemap is generated.*" [source](<https://www.sitemaps.org/protocol.html>)

would `FileDate` work?

Reply:
There is some miscommunication. 

We are talking about:
- `PublishDate` which is metadata from FronMatter
- `ModifiedDate` which is metadata we should implement in a future.
- `fileInfo.LastWriteTime` which is really last time the .md file was modified.

Now, what should be inside `lastmod` node?
Ideally `ModifiedDate` - it exactly matches the guidelines. But we don't have it (yet), so I am proposing to replace it with `PublishDate` as it gets closer to what `lastmod` means. I don't want to use `FileInfo` as doesn't hold the information about the content, but rather about file manipulation.